pkg/controllers/provisioning/provisioner.go

```diff
@@ -327,57 +325,21 @@ func (p *Provisioner) Launch(ctx context.Context, m *scheduler.Machine, opts ...
 	}
 	options := functional.ResolveOptions(opts...)
 
-	logging.FromContext(ctx).Infof("launching %s", m)
-	created, err := p.cloudProvider.Create(
-		logging.WithLogger(ctx, logging.FromContext(ctx).Named("cloudprovider")),
-		m.ToMachine(latest),
-	)
-	if err != nil {
-		return "", fmt.Errorf("creating cloud provider instance, %w", err)
-	}
-	k8sNode := &v1.Node{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:   created.Name,
-			Labels: created.Labels,
-		},
-		Spec: v1.NodeSpec{
-			ProviderID: created.Status.ProviderID,
-		},
-	}
-	ctx = logging.WithLogger(ctx, logging.FromContext(ctx).With("node", k8sNode.Name))
-
-	if err := mergo.Merge(k8sNode, m.ToNode()); err != nil {
-		return "", fmt.Errorf("merging cloud provider node, %w", err)
-	}
-	// ensure we clear out the status
-	k8sNode.Status = v1.NodeStatus{}
-
-	// Idempotently create a node. In rare cases, nodes can come online and
-	// self register before the controller is able to register a node object
-	// with the API server. In the common case, we create the node object
-	// ourselves to enforce the binding decision and enable images to be pulled
-	// before the node is fully Ready.
-	if _, err := p.coreV1Client.Nodes().Create(ctx, k8sNode, metav1.CreateOptions{}); err != nil {
-		if errors.IsAlreadyExists(err) {
-			logging.FromContext(ctx).Debugf("node already registered")
-		} else {
-			return "", fmt.Errorf("creating node %s, %w", k8sNode.Name, err)
-		}
-	}
-	if err := p.cluster.UpdateNode(ctx, k8sNode); err != nil {
-		return "", fmt.Errorf("updating cluster state, %w", err)
+	machine := m.ToMachine(latest)
+	if err := p.kubeClient.Create(ctx, machine); err != nil {
+		return "", err
```

Comment:
Think we should wrap here
```suggestion
		return "", fmt.Errof("creating machine object, %w", err)
```